### ISSUE TITLE
[Feature Request] Remote Sensors

### ISSUE THREAD

**Issue by AlchemyNZ** [exension/hwinfo-streamdeck#10]
I would like to be able to select to use remote sensors so I can report on my Gaming PC via my Streaming PC with streamdeck . I have the remote sensors showing in my HWinfo sensor list but are not visible in the plugin sensor list.

**Comment by taylorbohannan** [comment]
+1 I would love to have this feature as well. I would like to display the remote CPU/GPU/Disk/Network activity of my rack-mounted Plex server.

**Comment by TeeterTech** [comment]
+2 I set up the remote for this and then realized I couldn't get it to display on stream deck. My stream deck is connected to my stream rig I would love to view info of both my stream rig and gaming rig on it.

**Comment by shayne** [comment]
This could work. I've had requests to open-source the project. I want to, but per my agreement with the author of HWiNFO I need to keep the API closed source.

If I end up breaking it up into two separate projects this could work, one HWiNFO API and the other Stream Deck plugin.

Will look into.

**Comment by mySebbe** [comment]
any news on this feature ? I would love to have this function :)

**Comment by shayne** [comment]
Yeah I have finished breaking things apart and have a working plugin where the hardware service is running separate from the stream-deck plugin. I've tested remote/networked communication and it works. Need to work on network discovery and updating the user interface to support remote connections.

Getting close!

**Comment by TeeterTech** [comment]
Glad to hear there is progress on this. Thanks a lot man!

**Comment by RockNRollGeek** [comment]
Yet another +1 for this. Ended up here because I went to add my networked game pc sensors that are in my stream pc hwinfo and was confused why they weren't showing up. I think I saw another reply that was basically the same use case as I have: Stream decks are typically attached to a streaming PC. In a dual-pc setup, it's great to see your sensors on the deck, but not being able to monitor if your GPU/CPU on your game pc is going nuts is bad times. Obviously I can pull up hwinfo on my stream pc manually and instantly see my cpu/gpu/ram from both pc's right there in the same window, but of course the ideal is being able to quick glance over at my stream deck for that info.

**Comment by shayne** [comment]
Will release a beta version here in a week or two. Will prob require a bit of configuration but would be great to get feedback.

**Comment by fpsWillow** [comment]
Any news on this @shayne ?

Very curious on having this work for myself especially as I've had to ditch NZXT's horrible software for controlling temps in my PC

**Comment by RockNRollGeek** [comment]
*bumping* to see if there's any update on this

**Comment by vPeteWalker** [comment]
Hi again for February! Thank you!

**Comment by shayne** [comment]
Hey there- I hear ya. Hope to have some free time this month for this.

**Comment by mySebbe** [comment]
@shayne that would be so nice! much appreciated

**Comment by MirdautasVras** [comment]
@shayne Can't wait for that feature! Is the beta version already available somewhere? Thanks for your great work!!!

**Comment by revlisoft** [comment]
+1 as well.  Would love this feature especially with the extra buttons on the XL!  Thanks and keep up the great work!

**Comment by triggy-B** [comment]
+1 as well. Can't wait for the feature also. Was a little disappointed that this does not work but was relieved that the feature is already under construction. Keep up the good work!

**Comment by revlisoft** [comment]
Happy to help beta test and provide feedback with a streamdeck XL. :)

**Comment by shayne** [comment]
Setting time aside this coming week to work on this. I created a gitter chat for those who want to help test.

https://gitter.im/hwinfo-streamdeck/community

**Comment by adamjs83** [comment]
This can be done using Home Assistant. 

> I would like to be able to select to use remote sensors so I can report on my Gaming PC via my Streaming PC with streamdeck . I have the remote sensors showing in my HWinfo sensor list but are not visible in the plugin sensor list.

It is possible to get what you want working.  There is a great streamdeck plugin for Home Assistant which can get the info from OHM https://www.home-assistant.io/integrations/openhardwaremonitor/ on your gaming pc.

Gaming PC --> OHM --> Home Assistant --> Home Assistant Stream Deck Plugin --> Stream Deck

Here's a photo of the CPU and GPU temp on my home gaming pc from my stream deck at my office.

![stream](https://user-images.githubusercontent.com/2105363/130684560-2f6c1c84-1419-409b-b839-cf6987c8ec1d.jpg)

**Comment by RockNRollGeek** [comment]
Super excessive approach, like building a stick to reach out and flip a light switch when you can reacj your arm out and just flip it.   
  
  
  
 The info of the remote pc is already there in hwinfo on the stream deck pc. The sd addon is already reading the hwinfo info, just not that specific set of data. People shouldn’t have to install a home automation server on a separate piece of hardware/vm, go through all of it’s set up and configuration, then all the additional config to add the sensor info to said server, then the setup of getting the sd to talk to the home automation server, all just to read a couple pieces of data that are already available locally.
  

  
  
  
  

  
>

**Comment by adamjs83** [comment]
Agreed 100% but if somone wants this functionality today they could have it up and running under an hour. 

> Super excessive approach, like building a stick to reach out and flip a light switch when you can reacj your arm out and just flip it. The info of the remote pc is already there in hwinfo on the stream deck pc. The sd addon is already reading the hwinfo info, just not that specific set of data. People shouldn’t have to install a home automation server on a separate piece of hardware/vm, go through all of it’s set up and configuration, then all the additional config to add the sensor info to said server, then the setup of getting the sd to talk to the home automation server, all just to read a couple pieces of data that are already available locally.
> […](#)

**Comment by smcneece** [comment]
Did this go any place, love to be able to have 2 of my home servers on my StreamDeck.

**Comment by shayne** [comment]
The infrastructure is there and the code is now open-source. I'm looking for maintainers as I haven't the time to drive this project.

**Comment by zerychao** [comment]
You can do this:

1. Clone the repo, change every occurance of `com.exension.hwinfo` to something else, e.g., `com.exension.hwinfor0` in `*.go` files, change `C.HWiNFO_SENSORS_MAP_FILE_NAME2` to `"Global\\HWiNFO_SENS_SM2_REMOTE_0"` (line 68, with quotation marks) in the file `internal\hwinfo\shmem\shmem.go`, and compilie using:
   ```bash
   go build -o ./com.exension.hwinfo.sdPlugin/hwinfo.exe ./cmd/hwinfo_streamdeck_plugin
   go build -o ./com.exension.hwinfo.sdPlugin/hwinfo-plugin.exe ./cmd/hwinfo-plugin
   ```
   You can find the two compiled `.exe` files in the `com.exension.hwinfo.sdPlugin` folder.

2. Go to `%AppData%\Elgato\StreamDeck\Plugins\` folder, make a copy of `com.exension.hwinfo.sdPlugin` folder, rename it to `com.exension.hwinfor0.sdPlugin`, copy and overwrite the two `.exe` files, and change  `com.exension.hwinfo.reading`  (line 21), `HWiNFO` (line 9) to `com.exension.hwinfor0.reading`, `HWiNFO-R0` respectively in the `manifest.json` file.

3. Restart the Stream Deck software. Now you may see the `HWiNFO-R0` plugin and monitor the first remote machine you connected to in HWiNFO64. If you want to monitor e.g., the second remote machine, you can try changing the `0` mentioned above to `1` and repeat these steps. I didn't test this but it should probably work.

**Comment by ninbura** [comment]
Dang... was hoping this was implemented. 

Elgato just updated their iPad app with a 64x64 grid, was super interested in having all my stats for my 4 PCs in one place.

Would love to have this built in.

**Comment by Misiacy** [comment]
> You can do this:
> 
> 1. Clone the repo, change every occurance of `com.exension.hwinfo` to something else, e.g., `com.exension.hwinfor0` in `*.go` files, change `C.HWiNFO_SENSORS_MAP_FILE_NAME2` to `"Global\\HWiNFO_SENS_SM2_REMOTE_0"` (line 68, with quotation marks) in the file `internal\hwinfo\shmem\shmem.go`, and compile using:
>    ```shell
>    go build -o ./com.exension.hwinfo.sdPlugin/hwinfo.exe ./cmd/hwinfo_streamdeck_plugin
>    go build -o ./com.exension.hwinfo.sdPlugin/hwinfo-plugin.exe ./cmd/hwinfo-plugin
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    You can find the two compiled `.exe` files in the `com.exension.hwinfo.sdPlugin` folder.
> 2. Go to `%AppData%\Elgato\StreamDeck\Plugins\` folder, make a copy of `com.exension.hwinfo.sdPlugin` folder, rename it to `com.exension.hwinfor0.sdPlugin`, copy and overwrite the two `.exe` files, and change  `com.exension.hwinfo.reading`  (line 21), `HWiNFO` (line 9) to `com.exension.hwinfor0.reading`, `HWiNFO-R0` respectively in the `manifest.json` file.
> 3. Restart the Stream Deck software. Now you may see the `HWiNFO-R0` plugin and monitor the first remote machine you connected to in HWiNFO64. If you want to monitor e.g., the second remote machine, you can try changing the `0` mentioned above to `1` and repeat these steps. I didn't test this but it should probably work.

Hello 
Please upload the file

**Comment by moeilijk** [comment]
For remote machines use the sister plugin lhm-streamdeck (https://github.com/moeilijk/lhm-streamdeck); this plugin only reads local HWiNFO by design.

This project is under new maintenance. The plugin core has been completely replaced since this was filed: new engine, new configuration UI, rewritten HWiNFO bridge. The code this was reported against is gone, so we are closing the pre-v3 backlog and starting clean. If you still see this on v3.0, please open a new issue.
